.asf.yaml

```diff
@@ -44,37 +44,47 @@ github:
     main:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
+        required_conversation_resolution: true
     v1-10-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
     v2-0-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
     v2-1-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
     v2-2-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
     v2-3-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
     v2-4-stable:
       required_pull_request_reviews:
         required_approving_review_count: 1
+        required_linear_history: true
     v2-5-stable:
       required_pull_request_reviews:
-        required_approving_review_count: 1
+      required_approving_review_count: 1
+      required_linear_history: true
     v2-6-stable:
       required_pull_request_reviews:
-        required_approving_review_count: 1
+      required_approving_review_count: 1
+      required_linear_history: true
     v2-7-stable:
       required_pull_request_reviews:
-        required_approving_review_count: 1
+      required_approving_review_count: 1
+      required_linear_history: true
     v2-8-stable:
       required_pull_request_reviews:
-        required_approving_review_count: 1
-
+      required_approving_review_count: 1
+      required_linear_history: true
```

Comment:
How does this work for existing branches which may not have linear history?
Also, for newer vX-X-stable branches is it always possible to have linear histories during the release management activities? Asking just our of curiosity to understand this one.

Reply:
Protection works when you want to modify a branch - it does not go back to ancient history in the past. https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#require-linear-history

But yes - we always have exclusively linear history - we never allowed merges. in the "release" branches. It makes reasoning about history, cherry-picking, reverting about milion times easier and it also allows for very clear accountability (the change that gets merged is responsible for resolving all kind of conflicts. full stop) .

Reply:
The way how we do v2-x branches is that we always cherry-pick individual commits - there is no "merging" - just cherry-picking that creates a linear history for each branch beceuse those cherry-picked commits are separate "instances" of the same commit not merges (though they keep reference + PR# in the subjects so it's 100% trackable where they came from).